mautic/mautic
# Feature - Customize Preference Center

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/229
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/4776
| BC breaks? | N/A
| Deprecations? | N/A

#### Description:
This PR gives any Mautic admin user the ability to edit text labels, format and apply themes to Preference center landing page using the landing page builder.

#### Steps to test this PR:
1. Apply this PR and run migrations (`php app/console d:m:m`)
2. Create a Landing Page and toggle *Is Preference Center* (toggling this button shows/hides the new slots)

![image](https://user-images.githubusercontent.com/2924026/30647819-b38d1ff8-9dd9-11e7-9cd4-b041af1182d3.png)

3. Use the new slots in the builder to customize the page:

![image](https://user-images.githubusercontent.com/2924026/30647978-1d08c9d2-9dda-11e7-8070-fcd0dede2a7c.png)

4. Optionally, you can use tokens to insert the different slots:

![image](https://user-images.githubusercontent.com/2924026/30648108-7d14715a-9dda-11e7-94c5-3f3fd7e8cfa3.png)

5. To finish, don't forget to include a "save preferences" button, otherwise the preferences can't be saved:

![image](https://user-images.githubusercontent.com/2924026/30648227-c803079e-9dda-11e7-8354-a56cbb707a5c.png)

6. Save the page and create/edit an email
7. Select the Preference Center Page from the list:

![image](https://user-images.githubusercontent.com/2924026/30648867-4e44cc24-9ddc-11e7-98ba-1616c0ddbe59.png)

8. Send the email and upon receiving it click on the Unsubscribe link, the new preference center page will be displayed.

![image](https://user-images.githubusercontent.com/2924026/30649219-f3450e00-9ddc-11e7-90c9-6433a2ef9d01.png)

9. If no Preference Center page is selected in an email, the default page is displayed.

![image](https://user-images.githubusercontent.com/2924026/30649249-04a66bda-9ddd-11e7-9d59-27acd489ff35.png)
